src/server.cpp

```diff
@@ -82,8 +114,9 @@ namespace quicr {
                                              false);
 
         // Setup the function for the track handler to use to send objects with thread safety
+        std::weak_ptr weak_track_handler(track_handler);
         track_handler->publish_object_func_ =
-          [&, track_handler, subscribe_id = track_handler->GetSubscribeId(), cb = std::move(callback)](
+          [&, weak_track_handler, subscribe_id = track_handler->GetSubscribeId(), cb = std::move(callback)](
```

Comment:
Good catch! Subscribe id isn't used here either, should we remove it?

Reply:
Looks like it wasn't an issue to keep it, so I removed it.